modules/@apostrophecms/schema/index.js

```diff
@@ -465,6 +465,57 @@ module.exports = {
         });
       },
 
+      async isFieldRequired(field, destination) {
+        return field.requiredIf
+          ? evaluate(field.requiredIf, destination)
+          : field.required;
+
+        function evaluate(clause, destination) {
+          let result = true;
+
+          for (const [ key, val ] of Object.entries(clause)) {
+            const destinationKey = _.get(destination, key);
+
+            if (key === '$or') {
+              const results = val.map(clause => evaluate(clause, destination));
+              if (!results.some((value) => value)) {
+                result = false;
+                break;
+              }
+              continue;
+            } else if (val.$ne) {
+              // eslint-disable-next-line eqeqeq
+              if (val.$ne == destinationKey) {
+                result = false;
+                break;
+              }
+            }
+
+            if (val.min) {
```

Comment:
If we support it for `requiredIf`, you should add it to `if` too?

Reply:
All features should work equally for `if` and `requiredIf`.

Reply:
With as little code duplication as possible front and back end.

Reply:
done